qtawesome/icon_browser.py

```diff
@@ -0,0 +1,230 @@
+        copyCodeButton.clicked.connect(lambda: self._copyIconText(code=True))
+
+        lyt = QtWidgets.QVBoxLayout()
+        lyt.addWidget(self._lineEdit)
+        lyt.addWidget(self._listView)
+        lyt.addWidget(copyNameButton)
+        lyt.addWidget(copyCodeButton)
+
+        frame = QtWidgets.QFrame(self)
+        frame.setLayout(lyt)
+
+        self.setCentralWidget(frame)
+
+        QtWidgets.QShortcut(
+            QtGui.QKeySequence(QtCore.Qt.Key_Return),
+            self,
+            lambda: self._copyIconText(code=False)
+        )
+        QtWidgets.QShortcut(
+            QtGui.QKeySequence(QtCore.Qt.SHIFT | QtCore.Qt.Key_Return),
+            self,
+            lambda: self._copyIconText(code=True)
+        )
+
+        self._lineEdit.setFocus()
+
+        geo = self.geometry()
+        desktop = QtWidgets.QApplication.desktop()
+        screen = desktop.screenNumber(desktop.cursor().pos())
+        centerPoint = desktop.screenGeometry(screen).center()
+        geo.moveCenter(centerPoint)
+        self.setGeometry(geo)
+
+    def _contextMenu(self, pos):
+        """
+        Show a context menu for the list view at the provided position.
+
+        Parameters
+        ----------
+        pos : QtCore.QPoint
+        """
+        menu = QtWidgets.QMenu(self)
+
+        copyName = QtWidgets.QAction('Copy Name', self)
+        copyName.triggered.connect(lambda: self._copyIconText(code=False))
+        menu.addAction(copyName)
+
+        copyCode = QtWidgets.QAction('Copy Code', self)
+        copyCode.triggered.connect(lambda: self._copyIconText(code=True))
+        menu.addAction(copyCode)
+
+        menu.exec_(self.mapToGlobal(pos))
+
+    def _updateFilter(self):
+        """
+        Update the string used for filtering in the proxy model with the
+        current text from the line edit.
+        """
+        self._proxyModel.setFilterFixedString(self._lineEdit.text())
+
+    def _triggerDelayedUpdate(self):
+        """
+        Reset the timer used for committing the search term to the proxy model.
+        """
+        self._filterTimer.stop()
+        self._filterTimer.start()
+
+    def _triggerImmediateUpdate(self):
+        """
+        Stop the timer used for committing the search term and update the
+        proxy model immediately.
+        """
+        self._filterTimer.stop()
+        self._updateFilter()
+
+    def _copyIconText(self, code=False):
+        """
+        Copy the name of the currently selected icon to the clipboard.
+
+        Parameters
+        ----------
+        code : bool
+            If True, the python code required to instantiate the icon will
+            be copied to the clipboard.
+        """
+        indexes = self._listView.selectedIndexes()
+        if not indexes:
+            return
+
+        iconName = indexes[0].data()
+        if code is True:
+            content = "qtawesome.icon('%s')" % iconName
+        else:
+            content = "'%s'" % iconName
```

Comment:
It's weird to me that we copy the icon name with quotes in it.

Reply:
Sure - I can update the second case.  The first case will still exhibit the problem that you describe however.

Reply:
Could just solve both cases altogether by not supporting the copying as "code".

I think people are smart enough to read the README and figure out how to get an icon object. It's not that obscure.

Reply:
I think you're right - I've removed the option.  You can now only copy the name without any quotes around it.